### PR TITLE
PBTree: Fix NPE during concurrent creating and iterating schema

### PR DESCRIPTION
**Pull request by MarcosZyk**
## Description

1. Add stack trace logic during exception processing
2. Add page referent check and release logic during iterating B+ Tree pages to avoid wrong page eviction and deadlock.

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/schemaengine/schemaregion/mtree/impl/pbtree/flush/PBTreeFlushExecutor.java [hunk 1]
```diff
@@ -79,15 +79,15 @@ public void flushVolatileNodes() throws MetadataException {
         processFlushDatabase(databaseMNode);
       } catch (Exception e) {
         exceptions.add(e);
-        logger.error(e.getMessage());
+        logger.error(e.getMessage(), e);
```
Comment (Cpaulyz) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/schemaengine/schemaregion/mtree/impl/pbtree/flush/PBTreeFlushExecutor.java:82]:
```suggestion
        logger.warn(e.getMessage(), e);
```

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/schemaengine/schemaregion/mtree/impl/pbtree/flush/PBTreeFlushExecutor.java [hunk 2]
```diff
@@ -79,15 +79,15 @@ public void flushVolatileNodes() throws MetadataException {
...
       }
     }
     while (subtreeRoots.hasNext() && checkRemainToFlush()) {
       try {
         processFlushNonDatabase(subtreeRoots.next());
       } catch (Exception e) {
         exceptions.add(e);
-        logger.error(e.getMessage());
+        logger.error(e.getMessage(), e);
```
Comment (Cpaulyz) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/schemaengine/schemaregion/mtree/impl/pbtree/flush/PBTreeFlushExecutor.java:90]:
```suggestion
        logger.warn(e.getMessage(), e);
```

**Review by Cpaulyz — APPROVED** [review]
LGTM

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/schemaengine/schemaregion/mtree/impl/pbtree/flush/PBTreeFlushExecutor.java (hunk 1 above)
Comment (MarcosZyk) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/schemaengine/schemaregion/mtree/impl/pbtree/flush/PBTreeFlushExecutor.java]: Fixed.

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/schemaengine/schemaregion/mtree/impl/pbtree/flush/PBTreeFlushExecutor.java (hunk 2 above)
Comment (MarcosZyk) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/schemaengine/schemaregion/mtree/impl/pbtree/flush/PBTreeFlushExecutor.java]: Fixed.

**Pull request merged**